### PR TITLE
Role changes

### PR DESCRIPTION
**Pull request by e1Ru1o**
# Role Changes

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] New Agent
- [ ] Fix an agent error
- [ ] Update an agent logic/test/documentation
- [ ] Code style update (formatting, renaming)
- [ ] Other (please describe):

## New Agent PR readiness

- [x] `npm install` command works (i.e. package-lock is updated properly)
- [x] The `name` and `description` fields of `package.json` describe the agent properly
- [x] All the libraries listed in `package.json` are being used
- [x] All the development libraries are in `devDependencies` section of `package.json`
- [x] The agent README is updated
- [x] The agent README sufficiently describes what the agent does
- [x] All the alerts that the agent emits are listed in the README
- [x] The txn/block examples listed in the README generate the expected findings
- [x] Code is easy to follow and/or properly commented
- [x] Agent contains tests
- [x] All tests pass when running `npm test`
- [x] Tests are not making real network calls (i.e. network interaction is properly mocked)
- [x] Tests cover all the paths (i.e. negative & positive test cases)
- [x] When the tests complete, no warning/errors are shown
- [x] `npm start` works without errors when executed using a jsonRpcUrl for each of the networks specified in the README
- [x] The code does what the README describes
- [x] Network calls are properly being cached (if duplicated calls can occur)
- [x] A performance review has been done (e.g. `Promise.all` is used whenever possible)
- [x] The findings fields are filled out correctly
- [x] Findings `protocol` is specified (if the agent is for a specific one)
- [x] `metadata` of the findings contains all the information needed.
- [x] The SDK methods are being used appropriately (e.g. `filterLog`, `filterFunction`, `getEthersProvider`)
- [x] `ABI`s match the contracts used / `args` used from `LogDescriptions` exists in the `ABI`s
- [x] The agent follows Forta recommended [best practices](https://docs.forta.network/en/latest/best-practices/)
- [x] Agent can recover from failed error calls when they can fail (e.g. errors are being caught on calls that can fail)

### CODE REVIEW

File: Ondo-Finance/Role-Changes/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+
+## Supported Chains
+
+- Ethereum
+
+## Alerts
+
+- ONDO-1
+  - Fired when a transaction execute one or more of the above functions in Ondo Registry
```
Comment (RCantu92) [inline Ondo-Finance/Role-Changes/README.md:23]:
```suggestion
  - Fired when a transaction executes one or more of the above functions in the Ondo `Registry` contract.
```
Small correction and small suggestion.

File: Ondo-Finance/Role-Changes/src/agent.spec.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+      [DESC.GRANT_ROLE.toString()     , createAddress("0xe0a07"), registry, role("Role-4"), createAddress("0xabc04")],
+      [DESC.RENOUNCE_ROLE.toString()  , createAddress("0xe0a06"), createAddress("0x06"), role("Role-5"), createAddress("0xabc05")],
+      [DESC.REVOKE_ROLE.toString()    , createAddress("0xe0a05"), createAddress("0x07"), role("Role-6"), createAddress("0xabc06")],
+      [DESC.ADD_STRATEGIEST.toString(), createAddress("0xe0a04"), createAddress("0x08"), createAddress("0xfee"), "Strategist-Role-3"],
+      [DESC.GRANT_ROLE.toString()     , createAddress("0xe0a03"), createAddress("0x09"), role("Role-7"), createAddress("0xabc07")],
+      [DESC.RENOUNCE_ROLE.toString()  , createAddress("0xe0a02"), registry, role("Role-8"), createAddress("0xabc08")],
+      [DESC.REVOKE_ROLE.toString()    , createAddress("0xe0a01"), registry, role("Role-9"), createAddress("0xabc10")],
+      [DESC.NOT_REGISTRY.toString()   , createAddress("0xe0a123"), createAddress("0xdef1bad")],
+      [DESC.ENABLE_TOKENS.toString()  , createAddress("0xe0a321"), registry],
+    ];
```
Comment (RCantu92) [inline Ondo-Finance/Role-Changes/src/agent.spec.ts:63]: I think making an inline comment on which test cases are "wrong"/not supposed to be detected would help a bit with clarity.

**Review by RCantu92 — APPROVED** [review]
Great job as usual Lazaro. 👍 

Also, I think having `TEST_CASES` having the cases that are incorrect is great.

File: Ondo-Finance/Role-Changes/src/agent.spec.ts (hunk 2 above)
Comment (e1Ru1o) [reply Ondo-Finance/Role-Changes/src/agent.spec.ts:63]: ok, agree

**Review by yasmine9 — APPROVED** [review]
LGTM 👍

File: Ondo-Finance/Role-Changes/package.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    "push": "forta-agent push",
+    "disable": "forta-agent disable",
+    "enable": "forta-agent enable",
+    "keyfile": "forta-agent keyfile",
+    "format": "prettier --write \"src/**/*.ts\"",
+    "test": "jest"
+  },
+  "dependencies": {
+    "forta-agent": "^0.1.3",
+    "forta-agent-tools": "^2.0.5"
```
[the remark this answers is not in the document]
Comment (e1Ru1o) [reply Ondo-Finance/Role-Changes/package.json:27]: Sure!

**Pull request merged**